### DISCUSSION TITLE
Conflict with Rabbitmq package from vyuldashev

### DISCUSSION THREAD

**Discussion by rayan2014** [spatie/laravel-multitenancy#489] [Q&A]
I have already installed laravel multi tenancy, and i want to integrate Rabbitmq with my project, i ended up with conflicting with your pakage.
this is the original issue i opened on there repository https://github.com/vyuldashev/laravel-queue-rabbitmq/issues/569
any help ?

**Comment by rayan2014** [comment]
@telkins 
@benrolfe 
@freekmurze

**Comment by masterix21** [comment]
Feel free to create a repository to help us investigate the issue: better with a Dockerfile, to be ready to use. Thanks

**Comment by samppatiihonen** [comment]
I had same problem and this is how I solved it: 

1. Add following to `composer.json` to disable auto discovery.

```
// ...
  "extra": {
      "laravel": {
          "dont-discover": [
              "spatie/laravel-multitenancy",
              "vladimir-yuldashev/laravel-queue-rabbitmq"
          ]
      }
  },
// ...
```
2. Add providers manually to `/config/app.php`. RabbitMQ has to be before Multitenancy.

```
    'providers' => ServiceProvider::defaultProviders()->merge([
       // ...

        /** 3rd party providers */
        VladimirYuldashev\LaravelQueueRabbitMQ\LaravelQueueRabbitMQServiceProvider::class,
        Spatie\Multitenancy\MultitenancyServiceProvider::class,

    ])->toArray(),
```

3. Run `composer install`

Hope this works for you too.
